File: 013_LeetCode_1/3_reverse_integer.py

n = int(input("Enter Num: "))
sum = 0
while n>0:
    # rem = n%10
    # sum = sum*m + rem

    sum = sum*10 + n%10
    n //= 10
print(sum)



n = int(input("Enter Num: "))
rev = int(str(n)[::-1])
print(rev)

File: 013_LeetCode_1/3_reverse_integer.cpp
#include<iostream>
using namespace std;

int main()
{
    int num, ans=0, rem;
    cout<<"Enter integer: ";
    cin>>num;

    while(num>0)
    {
        rem = num%10;      
        num /= 10;
        ans = ans*10 + rem;
    }
    cout<<ans<<endl;
   
}

/* class Solution {
public:
int reverse(int x) {

int ans =0, rem;

while(x!=0) 
{
    rem = x%10; 
    x/=10;
    if(ans>INT_MAX/10 || ans<INT_MIN/10)
    return 0;
    ans = ans*10+rem; 
}

return ans;
}
}; */